events.go

```diff
@@ -110,6 +110,9 @@ func (e *eventRemote) Subscribe(ctx context.Context, filters ...string) (ch <-ch
 				Event:     ev.Event,
 			}:
 			case <-ctx.Done():
+				if cerr := ctx.Err(); cerr != context.Canceled {
+					errq <- cerr
```

Comment:
Need to check against nil to avoid a write of a nil error. Should probably be in the condition above.

Reply:
+1 why aren't we checking for a nil error?

Reply:
@cpuguy83 according to the doc https://golang.org/pkg/context/#Context, when the `ctx.Done()` has been closed, the `ctx.Err()` always returns the non-nil to explain why it is done. maybe it's caused by canceled or deadline exceeded.

It depends on the implementation of `context`. Add `nil` check is good here if there is some bug in context. But for now, I think it's ok.

Reply:
Ok I see the full context of this now (haha).

Reply:
LoL